### ISSUE TITLE
Nix build issue

### ISSUE THREAD

**Issue by bbarker** [typelead/eta#829]
I realize there has been some work on this (e.g. #781) but I still have issues when running `nix-shell -A eta-build-shell`:

```

Bindings/Utilities.hs:38:1: Warning: Tab character

Bindings/Utilities.hs:39:1: Warning: Tab character
Haddock coverage:
  67% (  2 /  3) in 'Bindings.Utilities'
  Missing documentation for:
    Module header
Documentation created: dist/doc/html/bindings-DSL/index.html,
dist/doc/html/bindings-DSL/bindings-DSL.txt
installing
Installing library in
/nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25/lib/ghc-7.10.3/bindings-DSL-1.0.25
Creating package registration file:
/nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25/lib/ghc-7.10.3/package.conf.d/bindings-DSL-1.0.25.conf
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25
shrinking /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25/lib/ghc-7.10.3/bindings-DSL-1.0.25/HSbindings-DSL-1.0.25-0NQhJ33cPs7A9J15aIsF4b.o
wrong ELF type
shrinking /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25/lib/ghc-7.10.3/bindings-DSL-1.0.25/libHSbindings-DSL-1.0.25-0NQhJ33cPs7A9J15aIsF4b-ghc7.10.3.so
strip is /nix/store/a3nk8z2i7m7wa3jdckgv710n7j3yx4b5-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25/lib 
patching script interpreter paths in /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25
checking for references to /tmp/nix-build-bindings-DSL-1.0.25.drv-0 in /nix/store/rdiishiailp8x4631s1qcxsjznfcmvwi-bindings-DSL-1.0.25...
wrong ELF type
shrinking RPATHs of ELF executables and libraries in /nix/store/fq7kx3vc19z97818m9y57y1xzp3v3zkc-bindings-DSL-1.0.25-doc
strip is /nix/store/a3nk8z2i7m7wa3jdckgv710n7j3yx4b5-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/fq7kx3vc19z97818m9y57y1xzp3v3zkc-bindings-DSL-1.0.25-doc
checking for references to /tmp/nix-build-bindings-DSL-1.0.25.drv-0 in /nix/store/fq7kx3vc19z97818m9y57y1xzp3v3zkc-bindings-DSL-1.0.25-doc...
building '/nix/store/897lxkbfk24f39z79h8j3b7dnvy44w8y-Cabal-1.24.2.0.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/zr5fi58zfb93wppln14gq0d0yzva8sz0-ghc-7.10.3.
unpacking sources
unpacking source archive /nix/store/4fqlanih86205bfy21qimh5aq881yxn4-Cabal-1.24.2.0.tar.gz
source root is Cabal-1.24.2.0
setting SOURCE_DATE_EPOCH to timestamp 1481168267 of file Cabal-1.24.2.0/tests/UnitTests.hs
patching sources
Replace Cabal file with edited version from http://hackage.haskell.org/package/Cabal-1.24.2.0/revision/2.cabal.
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-Cabal-1.24.2.0.drv-0/setup-package.conf.d -j8 -threaded -DMIN_VERSION_binary_0_8_0=1
[ 1 of 84] Compiling Distribution.Simple.Program.Internal ( Distribution/Simple/Program/Internal.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Simple/Program/Internal.o )
[ 2 of 84] Compiling Distribution.Compat.MonadFail ( Distribution/Compat/MonadFail.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/MonadFail.o )
[ 3 of 84] Compiling Distribution.Compat.GetShortPathName ( Distribution/Compat/GetShortPathName.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/GetShortPathName.o )
[ 4 of 84] Compiling Distribution.GetOpt ( Distribution/GetOpt.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/GetOpt.o )
[ 5 of 84] Compiling Distribution.Compat.Binary ( Distribution/Compat/Binary.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/Binary.o )
[ 6 of 84] Compiling Distribution.Compat.Semigroup ( Distribution/Compat/Semigroup.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/Semigroup.o )
[ 7 of 84] Compiling Distribution.Lex ( Distribution/Lex.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Lex.o )
[ 8 of 84] Compiling Distribution.TestSuite ( Distribution/TestSuite.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/TestSuite.o )
[ 9 of 84] Compiling Distribution.Simple.PreProcess.Unlit ( Distribution/Simple/PreProcess/Unlit.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Simple/PreProcess/Unlit.o )
[10 of 84] Compiling Distribution.Simple.CCompiler ( Distribution/Simple/CCompiler.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Simple/CCompiler.o )
[11 of 84] Compiling Distribution.PackageDescription.Utils ( Distribution/PackageDescription/Utils.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/PackageDescription/Utils.o )
[12 of 84] Compiling Distribution.Compat.ReadP ( Distribution/Compat/ReadP.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/ReadP.o )
[13 of 84] Compiling Distribution.ReadE ( Distribution/ReadE.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/ReadE.o )
[14 of 84] Compiling Distribution.Verbosity ( Distribution/Verbosity.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Verbosity.o )
[15 of 84] Compiling Distribution.Text ( Distribution/Text.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Text.o )
[16 of 84] Compiling Distribution.ModuleName ( Distribution/ModuleName.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/ModuleName.o )
[17 of 84] Compiling Distribution.System ( Distribution/System.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/System.o )
[18 of 84] Compiling Distribution.Version ( Distribution/Version.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Version.o )

Distribution/Version.hs:132:10:
    No instance for (Binary Version)
      arising from a use of ‘binary-0.7.5.0:Data.Binary.Class.$gdmput’
    In the expression: binary-0.7.5.0:Data.Binary.Class.$gdmput
    In an equation for ‘put’:
        put = binary-0.7.5.0:Data.Binary.Class.$gdmput
    In the instance declaration for ‘Binary VersionRange’
[22 of 84] Compiling Language.Haskell.Extension ( Language/Haskell/Extension.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Language/Haskell/Extension.o )
[26 of 84] Compiling Distribution.Compat.Exception ( Distribution/Compat/Exception.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/Exception.o )
[27 of 84] Compiling Distribution.Compat.Internal.TempFile ( Distribution/Compat/Internal/TempFile.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/Internal/TempFile.o )
[28 of 84] Compiling Distribution.Compat.CopyFile ( Distribution/Compat/CopyFile.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/CopyFile.o )
[41 of 84] Compiling Distribution.Compat.Environment ( Distribution/Compat/Environment.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/Environment.o )
[72 of 84] Compiling Distribution.Compat.CreatePipe ( Distribution/Compat/CreatePipe.hs, /tmp/nix-build-Cabal-1.24.2.0.drv-0/Distribution/Compat/CreatePipe.o )
builder for '/nix/store/897lxkbfk24f39z79h8j3b7dnvy44w8y-Cabal-1.24.2.0.drv' failed with exit code 1
building '/nix/store/9grv1j3p3j5nyypl7hg5qp2i1gihg0lr-bytestring-builder-0.10.8.1.0.drv'...
cannot build derivation '/nix/store/w6sbvfxxc14l5scwan73rhghm9dn28bk-etlas-1.3.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/gq0sqw763178l6pml7vrykrz3jh20rk6-eta-build-0.0.0.drv', '/nix/store/iy5jf2788i689zafpxq4ib3myy3993z9-eta-0.8.0.2.drv', '/nix/store/w6sbvfxxc14l5scwan73rhghm9dn28bk-etlas-1.3.0.0.drv', '/nix/store/y4bfycm1qrlbvjj944ll
22a916hlrg98-eta-pkg-0.8.0.2.drv' failed                                                                                                                                                                                                          
brandon@brandon-750-170se-DevContainer:~/workspace/eta

```
I did this from the latest version of stable branch in the eta repo.

**Comment by rahulmutt** [comment]
@bbarker Can you follow along with #827?

**Comment by bbarker** [comment]
@rahulmutt sorry, I shamefully failed to do a proper search this time. The good news is the fix associated with #827 seems to have worked for me on master (2e7aefd7b0da16725c4977a3dadf041303fe5d40)!
